internal/core/scale.go

```diff
@@ -34,16 +34,36 @@ func encodeProof(dnmp eth.DoubleNodeWithMerkleProof) string {
 	return hex
 }
 
+func lenToBytes(b []uint8, len int) ([]uint8, int) {
+	if len < 255 {
+		b = append(b, uint8(len))
+		len = 0
+		return b, len
+	}
+
+	b = append(b, uint8(len/0xff))
+	len = len % 0xff
+	return lenToBytes(b, len)
+}
+
+func lenToHex(len int) string {
+	b, _ := lenToBytes([]uint8{}, len*4)
+	return hex.EncodeToString(b)
+}
+
 // Encode Darwinia Eth Header
 func encodeDarwiniaEthHeader(header eth.DarwiniaEthHeader) string {
+	hb, _ := hex.DecodeString(header.ExtraData[2:])
+	len := lenToHex(len(hb))
+
 	hex := "0x"
 	hex += header.ParentHash[2:]
 	hex += encodeUint(header.TimeStamp, 64)
 	hex += encodeUint(header.Number, 64)
 	hex += strings.ToLower(header.Author[2:])
 	hex += header.TransactionsRoot[2:]
 	hex += header.UnclesHash[2:]
-	hex += "7c"
+	hex += len
```

Comment:
Here is the BUG in the last version of scale codec, used a fixed vector length.

Reply:
```
# Run shadow with rust logs
$ RUST_LOG=mmr shadow run --fetch

# Post the proposal API
$ curl -H "content-type: application/json" --request POST --data '{ "members": [0], "last_leaf": 3, "format": "codec" }' http://localhost:3001/api/v1/proposal

# The swagger docs
http://localhost:3001/swagger/index.html
```

Reply:
> Here is the BUG in the last version of scale codec, used a fixed vector length.

For "last version of scale codec", you mean some 3rd party libraries?

Reply:
Last time write the scale codec for `off-chain worker`, but not in production, this `7c` -> `len` means the length of `extra_data` field in eth header, the hard code `7c` is only the `extra_data` of block number `8997676`.